components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -2,55 +2,99 @@ import styled from "styled-components";
 import { useRef, useState, useEffect } from "react";
 
 import ListContainer from "../ListContainer";
+import InputDropDown from "./InputDropDown";
+import { getAllItemsFromDB } from "/services/db.js";
 
 export default function ShoppingListEditor({ items, onDelete, onAdd }) {
+  const [allItems, setAllItems] = useState(getAllItemsFromDB);
+  const [itemInput, setItemInput] = useState("");
+  const [dropDownItems, setDropDownItems] = useState([]);
+  const [dropDownReset, setDropDownReset] = useState(false);
   const inputRef = useRef();
 
+  //set focus on item input after page load
   useEffect(() => {
     inputRef.current.focus();
-  }, []); //set focus on item input after page load
+  }, []);
 
-  function submitForm(event) {
-    event.preventDefault();
-    const form = event.target;
-    const formData = new FormData(form);
-    const itemName = Object.fromEntries(formData).item;
+  //open drop down when typing into input field
+  useEffect(() => {
+    triggerDropDown();
+  }, [itemInput]);
```

Comment:
by using useEffect you double the rerenders for each change in the input. Why not calling triggerDropDown directly in the onChange function of the input?